addons/crm/models/crm_team.py

```diff
@@ -113,12 +146,329 @@ def _alias_get_creation_values(self):
         if self.id:
             if not self.use_leads and not self.use_opportunities:
                 values['alias_name'] = False
-            values['alias_defaults'] = defaults = ast.literal_eval(self.alias_defaults or "{}")
+            values['alias_defaults'] = defaults = literal_eval(self.alias_defaults or "{}")
             has_group_use_lead = self.env.user.has_group('crm.group_use_lead')
             defaults['type'] = 'lead' if has_group_use_lead and self.use_leads else 'opportunity'
             defaults['team_id'] = self.id
         return values
 
+    # ------------------------------------------------------------
+    # LEAD ASSIGNMENT
+    # ------------------------------------------------------------
+
+    @api.model
+    def _cron_assign_leads(self):
+        """ Cron method assigning leads. It is based on cron configuration to
+        know parameters of assignment to perform. """
```

Comment:
Would be nice to improve a bit with an example:
> e.g: CRON is setup to run three times every month: 2 * 3 * 30, passed work_days will be 180

That also means that if the CRON runs MORE, we assign MORE every time?
Is that intended?

Side note: also noticed that assignment will fail if work_days is more than 30? Shouldn't that be handled?
What am I missing?

Reply:
An internal_number of 3 means every 3 months and not 3 times per month, I need more sleep.
You can disregard that part:
> That also means that if the CRON runs MORE, we assign MORE every time?
> Is that intended?

The rest is still valid though I think.